x/fees/keeper/evm_hooks.go

```diff
@@ -0,0 +1,74 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// Hooks wrapper struct for fees keeper
+type Hooks struct {
+	k Keeper
+}
+
+var _ evmtypes.EvmHooks = Hooks{}
+
+// Return the wrapper struct
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k}
+}
+
+// PostTxProcessing implements EvmHooks.PostTxProcessing. After each successful
+// interaction with a registered contract, the contract deployer receives
+// a share from the transaction fees paid by the user.
+func (h Hooks) PostTxProcessing(ctx sdk.Context, msg core.Message, receipt *ethtypes.Receipt) error {
+	// check if the fees are globally enabled
+	params := h.k.GetParams(ctx)
+	if !params.EnableFees {
+		return nil
+	}
+
+	contract := msg.To()
+	// if the contract is not registered to receive fees, do nothing
+	if contract == nil || !h.k.IsFeeRegistered(ctx, *contract) {
+		return nil
+	}
+
+	cfg, err := h.k.evmKeeper.EVMConfig(ctx)
+	if err != nil {
+		return err
```

Comment:
In `ApplyTransaction` there is an error check for the EVMConfig before the `PostTxProcessing` call
https://github.com/tharsis/ethermint/blob/ea3c16ed7a3af89ea5ba9af2071a54b2cfb0a551/x/evm/keeper/state_transition.go#L199-L202
So, in `x/fees/keeper/evm_hooks.go`, getting the `EVMConfig` should never error.
In this case, is it better to return `nil` instead of an error, to not revert the entire transaction? I think the transaction is more important than the fees.

Reply:
I see, thanks for the context. Let's return `nil` and add that as a comment:

```go
// shouldn't occur: error is already checked in EVM's ApplyTransaction
```